lib/sdf-server/src/server/service/v2/func/create_func.rs

```diff
@@ -0,0 +1,265 @@
+use axum::{
+    extract::{OriginalUri, Path},
+    response::IntoResponse,
+    Json,
+};
+use serde::{Deserialize, Serialize};
+
+use dal::{
+    func::{
+        authoring::FuncAuthoringClient,
+        binding::{
+            AttributeArgumentBinding, AttributeFuncArgumentSource, AttributeFuncDestination,
+            EventualParent,
+        },
+        FuncKind,
+    },
+    schema::variant::leaves::{LeafInputLocation, LeafKind},
+    ChangeSet, ChangeSetId, Func, WorkspacePk, WsEvent,
+};
+use si_frontend_types::{self as frontend_types, FuncBinding, FuncCode, FuncSummary};
+
+use crate::server::{
+    extract::{AccessBuilder, HandlerContext, PosthogClient},
+    tracking::track,
+};
+
+use super::{get_code_response, get_types, FuncAPIError, FuncAPIResult};
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateFuncRequest {
+    name: Option<String>,
+    display_name: Option<String>,
+    description: Option<String>,
+    binding: frontend_types::FuncBinding,
+    kind: FuncKind,
+}
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateFuncResponse {
+    summary: FuncSummary,
+    code: FuncCode,
+}
+
+pub async fn create_func(
+    HandlerContext(builder): HandlerContext,
+    AccessBuilder(access_builder): AccessBuilder,
+    PosthogClient(posthog_client): PosthogClient,
+    OriginalUri(original_uri): OriginalUri,
+    Path((_workspace_pk, change_set_id)): Path<(WorkspacePk, ChangeSetId)>,
+    Json(request): Json<CreateFuncRequest>,
+) -> FuncAPIResult<impl IntoResponse> {
+    let mut ctx = builder
+        .build(access_builder.build(change_set_id.into()))
+        .await?;
+    if let Some(name) = request.name.as_deref() {
+        if dal::func::is_intrinsic(name)
+            || ["si:resourcePayloadToValue", "si:normalizeToArray"].contains(&name)
+        {
+            return Err(FuncAPIError::FuncNameReserved(name.into()));
+        }
+    }
+
+    let force_change_set_id = ChangeSet::force_new(&mut ctx).await?;
+
+    let created_func_response = match request.kind {
```

Comment:
This is more of an architectural suggestion, but i see that every branch that returns something on this match ends up running a variation of:
```
                CreateFuncResponse {
                    summary: func.into_frontend_type(&ctx).await?,
                    code: get_code_response(&ctx, created_func.id).await?,
                }
```

which makes me think that it would make senes for the match to return a Func (or even just the func id) instead of that type